File: countPrimes.js
var countPrimes = function (n) {
  if (n < 2) return 0;
  let primes = 0;

  let newArr = new Array(n).fill(false);
  let filler = 2;

  while (filler < newArr.length){
    newArr[n] = true;
    filler += 2;
  }

  for (let i = 3; i < newArr.length; i++) {
    if (newArr[i]) {
      primes += 1
    } else {
      primes += isPrime(i)
    }
  }

  return primes;
};

const isPrime = (num) => {
    if (num < 2) return 0;
    if (num === 2) return 1;
    for (let i = 2; i < num; i++) {
        if (num % i === 0) return 0; 
    }
    return 1;
}

console.log(countPrimes(100000) )

File: ruby/count_primes.rb
def count_primes(n)
  count = 0;
  first_h = 0;
  rest = n % 100;

  if n < 100
    (0...n).each do |i|
        count += 1 if is_prime(i)
    end
    return count;
  end

  if n > 100
    (0...100).each do |i|
        first_h += 1 if is_prime(i)
    end

    (n-rest...n).each do |i|
        count += 1 if is_prime(i)
    end
  end
  count += (first_h * (n/100))
  return count;
end

def is_prime(num)
    return false if num <= 1
    return true if num == 2
     i = 2
    while i < num
        return false if num % i == 0
        i += 1
    end
return true
end

p count_primes(499979)
#